### PR TITLE
Skip table consistency check in recovery mode

### PR DESCRIPTION
**Pull request by hamadodene**

### CODE REVIEW

File: herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecovery.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.concurrent.atomic.AtomicInteger;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+
+/**
+ *
+ * @author hamado
+ */
+public class ConsistencyCheckDuringRecovery extends ReplicatedLogtestcase{
```
Comment (eolivelli) [inline herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecovery.java:45]: Test classes names must end with Test

File: herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java [hunk 2]
```diff
@@ -98,6 +98,7 @@
 
     private volatile boolean closed = false;
     private volatile boolean failed = false;
+    private final boolean skipTableConsistencyCheckDuringRecovery = true;
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java:101]: Why do you need this flag?

File: herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecovery.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            manager1.waitForTablespace("t2", 10000);
+        }
+        
+        try (DBManager manager1 = startDBManager("node2")) {
+            //restart follower  node
+            manager1.waitForTablespace("t2", 1000);
+        }
+        
+        
+                //Expected 2 call for createChecksum (node1 and node2)
```
Comment (eolivelli) [inline herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecovery.java:101]: 2 or 3?

File: herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java (hunk 2 above)
Comment (hamadodene) [reply herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java:101]: `                                    if (number.after(snapshotSequenceNumber) ) {
                                        if (LOGGER.isLoggable(Level.FINEST)) {
                                            LOGGER.log(Level.FINEST, "rec " + tableSpaceName + " #" + localEntryCount + " {0}, {1}",
                                                    new Object[]{number, statusEdit});
                                        }
                                        consumer.accept(number, statusEdit);
                                    } `

THe idea was to skip at this point if the consistency entry was received

Comment (hamadodene) [reply herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java:101]: I think I can use entry.type

File: herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecoveryTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ *
+ * @author hamado
+ */
+public class ConsistencyCheckDuringRecoveryTest extends ReplicatedLogtestcase {
+
+    final String tableName = "table1";
+    final String tableSpaceName = "t2";
+
+    @Test
+    public void consistencyCheckReplicaTest() throws Exception {
```
Comment (eolivelli) [inline herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecoveryTest.java:51]: Is this test failing without your fix?

File: herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java [hunk 5]
```diff
@@ -616,7 +616,7 @@ public void recovery(
                                     lastLedgerId = ledgerId;
                                     currentLedgerId = ledgerId;
                                     lastSequenceNumber.set(entryId);
-                                    if (number.after(snapshotSequenceNumber)) {
+                                    if (number.after(snapshotSequenceNumber) && statusEdit.type != LogEntryType.TABLE_CONSISTENCY_CHECK) {
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java:619]: This is not the right place.
I think it should be done in TableSpaceManager 
this class should not be aware of the semantics of entries

File: herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecoveryTest.java (hunk 4 above)
Comment (hamadodene) [reply herddb-core/src/test/java/herddb/data/consistency/ConsistencyCheckDuringRecoveryTest.java:51]: yes

File: herddb-core/src/main/java/herddb/core/TableSpaceManager.java [hunk 6]
```diff
@@ -2068,7 +2068,13 @@ public void accept(LogSequenceNumber t, LogEntry u) {
                 throw new RuntimeException("System was requested to stop, aborting recovery at " + t);
             }
             try {
-                apply(new CommitLogResult(t, false, true), u, true);
+                if (u.type != LogEntryType.TABLE_CONSISTENCY_CHECK) {
+                    apply(new CommitLogResult(t, false, true), u, true);
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/core/TableSpaceManager.java:2072]: why don't you push this check into "apply" ?

Comment (hamadodene) [reply herddb-core/src/main/java/herddb/core/TableSpaceManager.java:2072]: right

**Pull request merged**